src/containers/community/CommunityJoin.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+import { connect } from 'react-redux'
+const { func, bool } = React.PropTypes
+import { get, isEmpty } from 'lodash'
+import { joinCommunityWithCode, navigate, resetCommunityValidation, validateCommunityAttribute } from '../../actions'
+import cx from 'classnames'
+
+@connect(({communityValidation}) => ({
+  codeInvalid: !get(communityValidation, 'beta_access_code.exists')
+}))
+export default class CommunityJoin extends React.Component {
+  static propTypes = {
+    dispatch: func,
+    codeInvalid: bool
+  }
+
+  codeChanged = event => {
+    let { dispatch } = this.props
+    let { value } = event.target
+    if (isEmpty(value.trim())) {
+      dispatch(resetCommunityValidation('beta_access_code'))
+    } else {
+      dispatch(validateCommunityAttribute('beta_access_code', value, 'exists'))
+    }
+  }
+
+  submit = () => {
+    let { dispatch, codeInvalid } = this.props
+
+    if (codeInvalid) {
+      return
+    }
+    let code = this.refs.code.value
+    dispatch(joinCommunityWithCode(code))
+    .then(action => {
+      if (action.error) {
+        console.log('TODO - display error message')
```

Comment:
all the errors are being stored in the `errors` subtree of state, so you can use that to get an error message into props via `connect` -- follow the error presentation style of Login and Signup